src/main.cpp

```diff
@@ -125,7 +172,9 @@ int main(const int argc, const char *const *const argv) {
         return false;
     };
 
-    std::copy_if(QtModules.begin(), QtModules.end(), std::back_inserter(foundQtModules),
+    const std::vector<QtModule> *qtmodules = qtMajorVersion == 6 ? &Qt6Modules : &Qt5Modules;
```

Comment:
Please, no plain pointers. You could use a reference here as well. Or at least a smart pointer. Ideally, this decision is not taken here anyway. A separate function would be cleaner, as it gives you greater freedom when writing such code.

```cpp
const std::vector<QtModule>& getQtModules() {
    if (qtMajorVersion == 5) {
        return Qt5Modules;
    } else if (qtMajorVersion == &) {
        return Qt6Modules;
    }

    throw some_exception;
}

[...]

const std::vector<QtModule>& qtModules = getQtModules();
```

Reply:
Ok, changes applied.